modules/rosa-sts-cluster-terraform.adoc

```diff
@@ -0,0 +1,516 @@
+    additional_trust_bundle = optional(string) # a string contains contains a PEM-encoded X.509 certificate bundle that will be added to the nodes' trusted certificate store.
+    no_proxy                = optional(string) # no proxy
+  })
+}
+
+#Private link cluster Info
+variable "enable_private_link" {
+  type        = bool
+  description = "This enables private link"
+  default     = false
+}
+
+#VPC Info
+variable "create_vpc" {
+  type        = bool
+  description = "Create custom VPC for ROSA cluster"
+}
+
+variable "vpc_name" {
+  type        = string
+  description = "VPC Name"
+  default     = null
+}
+
+variable "vpc_cidr_block" {
+  type        = string
+  description = "value of the CIDR block to use for the VPC"
+  default     = "10.0.0.0/16"
+}
+
+variable "private_subnet_cidrs" {
+  type        = list(any)
+  description = "The CIDR blocks to use for the private subnets"
+  default     = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24"]
+}
+
+variable "public_subnet_cidrs" {
+  type        = list(any)
+  description = "The CIDR blocks to use for the public subnets"
+  default     = ["10.0.101.0/24", "10.0.102.0/24", "10.0.103.0/24"]
+}
+
+variable "single_nat_gateway" {
+  type        = bool
+  description = "Single NAT or per NAT for subnet"
+  default     = false
+}
+
+#AWS Info
+variable "aws_region" {
+  type    = string
+  default = "us-east-2"
+}
+
+variable "availability_zones" {
+  type        = list(any)
+  description = "The availability zones to use for the cluster"
+  default     = ["us-east-2a", "us-east-2b", "us-east-2c"]
+}
+
+variable "installer_role_arn" {
+  description = "STS Role ARN with get secrets permission, relevant only for unmanaged OIDC config"
+  type        = string
+  default     = null
+}
+
+variable "private_subnet_ids" {
+  type        = list(any)
+  description = "VPC private subnets IDs for ROSA Cluster"
+  default     = []
+}
+EOF
+----
+
+.. Run the following command to create your `vpc.tf` file:
++
+[NOTE]
+====
+This guide does not use a VPC. You can change `create_vpc` to `true` in the `terraform.tfvars` file if you want to use a VPC.
+====
++
+[source,terminal]
+----
+$ cat<<-EOF>vpc.tf
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "5.1.2"
+
+  count = var.create_vpc ? 1 : 0
+  name  = var.vpc_name
+  cidr  = var.vpc_cidr_block
+
+  azs             = var.availability_zones
+  private_subnets = var.private_subnet_cidrs
+  public_subnets  = var.public_subnet_cidrs
+
+  enable_nat_gateway   = true
+  single_nat_gateway   = var.single_nat_gateway
+  enable_dns_hostnames = true
+  enable_dns_support   = true
+
+  tags = var.additional_tags
+}
+EOF
+----
+. Create a `terraform.tfvars` file by running the following command:
++
+[source,terminal]
+----
+$ cat<<-EOF>terraform.tfvars
+#AWS Info
+aws_region="us-east-1"
+availability_zones = ["us-east-1a"]
+redhat_aws_account_id="<AWS_account_ID>"
+
+# Create a VPC?
+create_vpc="false"
```

Comment:
If by default false, I would add a note saying that the user has to provide the correct `var. private_subnet_ids`, which is currently set to `[]` in the variables.tf

Reply:
Either remove the default from variables.tf so the user needs to input it, or set create_vpc to true by default in my opinion